### PR TITLE
Make tests run on moto server

### PR DESCRIPTION
**Pull request by martindurant**
- [x] Tests added / passed
- [x] Passes `black dask` / `flake8 dask`

Like https://github.com/pandas-dev/pandas/pull/35655

### CODE REVIEW

File: continuous_integration/environment-3.6.yaml [hunk 1]
```diff
@@ -18,7 +19,7 @@ dependencies:
   # - fastparquet
   - h5py
   - pytables
-  - fsspec==0.6.0
+  - fsspec>=0.7.4
```
Comment (jsignell) [inline continuous_integration/environment-3.6.yaml:22]: Do the other envs not need to pin a min fsspec like this?

Comment (martindurant) [reply continuous_integration/environment-3.6.yaml:22]: All the others had ">=", so would have picked up the latest. I don't know where the failure is coming from, doesn't happen for me even on windows :|

**Comment by martindurant** [comment]
Tests do pass, failure is in teardown (see https://github.com/goanpeca/setup-miniconda/issues/63 )

**Comment by martindurant** [comment]
@dask/maintenance - this is ready, except for the note above (which is not related to the PR content)

**Comment by martindurant** [comment]
Will reset to released GH tag, once available

**Pull request merged**